lib/logic.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: false
+
+def bin_converter(state)
+  state.reduce(0) { |acc, value| acc * 2 + value }
+end
+
+WINNER_STATES = [
+
+  [1, 0, 0,
+   1, 0, 0,
+   1, 0, 0],
+
+  [0, 1, 0,
+   0, 1, 0,
+   0, 1, 0],
+
+  [0, 0, 1,
+   0, 0, 1,
+   0, 0, 1],
+
+  [1, 1, 1,
+   0, 0, 0,
+   0, 0, 0],
+
+  [0, 0, 0,
+   1, 1, 1,
+   0, 0, 0],
+
+  [0, 0, 0,
+   0, 0, 0,
+   1, 1, 1],
+
+  [1, 0, 0,
+   0, 1, 0,
+   0, 0, 1],
+
+  [0, 0, 1,
+   0, 1, 0,
+   1, 0, 0]
+
+].map { |state| bin_converter(state) }
+
```

Comment:
About `bin_converter()` and `WINNER_STATES`, there shouldn't be the logic elements outside from class